src/main/java/core/basesyntax/db/FruitDaoImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.db;
+
+import core.basesyntax.db.storage.Storage;
+import java.util.Map;
+
+public class FruitDaoImpl implements FruitDao {
+    @Override
+    public void add(String fruit, int quantity) {
+        if (!Storage.fruits.containsKey(fruit)) {
+            Storage.fruits.put(fruit,quantity);
+            return;
+        }
+        Storage.fruits.compute(fruit, (k, oldQuantity) -> oldQuantity + quantity);
+    }
+
+    @Override
+    public void subtract(String fruit, int quantity) {
+        int oldQuantity = Storage.fruits.get(fruit);
```

Comment:
Consider checking if the fruit exists in storage before attempting to get its quantity. This will prevent a potential `NullPointerException` if the fruit is not present.